src/main/java/iudx/catalogue/server/apiserver/RelationshipApis.java

```diff
@@ -65,37 +66,54 @@ public void listRelationshipHandler(RoutingContext routingContext) {
       String itemType = request.getParam(REL_KEY);
 
       if (!id.isBlank() && ITEMS_KEY.contains(itemType)) {
-        requestBody.put(ID, id);
-        requestBody.put(INSTANCE, instanceID);
-        requestBody.put(RELATIONSHIP, itemType);
-
-        /*
-         * Request database service with requestBody for listing resource relationship
-         */
-        dbService.listRelationship(requestBody, dbhandler -> {
-          if (dbhandler.succeeded()) {
-            LOGGER.info("Success: Retrieved relationships of " + itemType);
-            response.setStatusCode(200)
-                    .end(dbhandler.result().toString());
-          } else if (dbhandler.failed()) {
-            LOGGER.error("Fail: Issue in listing relationship;" + dbhandler.cause().getMessage());
+        requestBody = QueryMapper.map2Json(queryParameters);
+
+        if (requestBody != null) {
```

Comment:
Can we fail first instead of having it nested inside?
For e.g
If fail -> return
<code>

instead of
if !fail 
   <code>
else
   <code>